app/jobs/regular/change_display_name.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChangeDisplayName < ::Jobs::Base
+    sidekiq_options queue: "low"
+
+    def execute(args)
+      @user = User.find_by(id: args[:user_id])
+
+      return unless user.present?
+
+      # We need to account for the case where the instance allows
+      # name to be empty by falling back to username.
+      @old_display_name = (args[:old_name].presence || user.username).unicode_normalize
+      @new_display_name = (args[:new_name].presence || user.username).unicode_normalize
+
+      @quote_rewriter = QuoteRewriter.new(user.id)
+
+      update_posts
+      update_revisions
+    end
+
+    private
+
+    attr_reader :user, :old_display_name, :new_display_name, :quote_rewriter
+
+    def update_posts
+      Post
+        .with_deleted
+        .joins(quoted("posts.id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |post| update_post(post) }
```

Comment:
Since we're likely to be dealing with alot of posts here. I wonder if we can just pluck the columns we required, update the raws and the cooked and do a bulk update for each batch instead. Otherwise, I worry that this job will end up hammering the DB with many queries in a short span of time.

Reply:
I'm going to push this to a TODO, partly because the code is lift-and-shift from username changing, where it hasn't caused any issue so far, and partly because I'm concerned the longer we keep records in memory before saving the bigger the chance we overwrite other changes that may have happened in the interim. 🙏

Reply:
Ah. I figured it out now. There are two sides to it. One side is the side that's doing the quoting, and the other side is the post that gets quoted, and `quoted_posts` acts as the join table.

We need the quoting post where the quoted post user is a certain ID, or in other words we need both sides of posts, hence the extra join,